gnd/src/main/java/com/google/android/gnd/persistence/remote/RemoteDataEvent.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gnd.persistence.remote
+
+import com.google.android.gnd.rx.ValueOrError
+
+/**
+ * An event returned by the remote data store indicating either an entity was successfully loaded,
+ * or that it was modified or removed in the remote data store.
+ *
+ * @param <T> the type of entity being loaded, modified, or removed.
+</T> */
+class RemoteDataEvent<T> private constructor(
```

Comment:
Once we fully convert to Kotlin, we should remove this in favor of [kotlin.Result](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/) and just keep the ENUM. Just an FYI. I don't think we can do this until we don't have any Java callers.

Reply:
Added a TODO